packages/inference/src/lib/makeRequestOptions.ts

```diff
@@ -35,6 +43,116 @@ export async function makeRequestOptions(
+
+	const { includeCredentials, taskHint, chatCompletion, signal } = options ?? {};
+
+	if (endpointUrl && provider !== "hf-inference") {
+		throw new Error(`Cannot use endpointUrl with a third-party provider.`);
+	}
+	if (maybeModel && isUrl(maybeModel)) {
+		throw new Error(`Model URLs are no longer supported. Use endpointUrl instead.`);
+	}
+	if (!maybeModel && !taskHint) {
+		throw new Error("No model provided, and no task has been specified.");
+	}
+	if (!providerConfig) {
+		throw new Error(`No provider config found for provider ${provider}`);
+	}
+	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+	const hfModel = maybeModel ?? (await loadDefaultModel(taskHint!));
+	const model = await getProviderModelId({ model: hfModel, provider }, args, {
+		taskHint,
+		chatCompletion,
+		fetch: options?.fetch,
+	});
+
+	/// If accessToken is passed, it should take precedence over includeCredentials
+	const authMethod = accessToken
+		? accessToken.startsWith("hf_")
+			? "hf-token"
+			: "provider-key"
+		: includeCredentials === "include"
+		  ? "credentials-include"
+		  : "none";
+
+	// Make URL
+	const url = endpointUrl
+		? chatCompletion
+			? endpointUrl + `/v1/chat/completions`
+			: endpointUrl
+		: providerConfig.makeUrl({
+				baseUrl:
+					authMethod !== "provider-key"
+						? HF_HUB_INFERENCE_PROXY_TEMPLATE.replace("{{PROVIDER}}", provider)
+						: providerConfig.baseUrl,
+				model,
+				taskHint,
+				chatCompletion,
+		  });
+
+	// Make headers
+	const binary = "data" in args && !!args.data;
+	const headers = providerConfig.makeHeaders({
+		accessToken,
+		authMethod,
+		binary,
+	});
+
+	// Add user-agent to headers
+	// e.g. @huggingface/inference/3.1.3
+	const ownUserAgent = `${packageName}/${packageVersion}`;
+	const userAgent = [ownUserAgent, typeof navigator !== "undefined" ? navigator.userAgent : undefined]
+		.filter((x) => x !== undefined)
+		.join(" ");
+	headers["User-Agent"] = userAgent;
+
+	// Make body
+	const body = binary
+		? args.data
+		: JSON.stringify(
+				providerConfig.makeBody({
+					args: remainingArgs as Record<string, unknown>,
+					model,
+					taskHint,
+					chatCompletion,
+				})
+		  );
+
+	/**
+	 * For edge runtimes, leave 'credentials' undefined, otherwise cloudflare workers will error
+	 */
+	let credentials: RequestCredentials | undefined;
+	if (typeof includeCredentials === "string") {
+		credentials = includeCredentials as RequestCredentials;
+	} else if (includeCredentials === true) {
+		credentials = "include";
+	}
+
+	const info: RequestInit = {
+		headers,
+		method: "POST",
+		body,
+		...(credentials ? { credentials } : undefined),
+		signal,
+	};
+
+	return { url, info };
+}
+
+export async function makeRequestOptionsLegacy(
+	args: RequestArgs & {
+		data?: Blob | ArrayBuffer;
+		stream?: boolean;
+	},
+	options?: Options & {
+		/** To load default model if needed */
+		taskHint?: InferenceTask;
+		chatCompletion?: boolean;
+	}
```

Comment:
Do you think it would be possible to merge the two `makeRequestOptions` functions?

Reply:
`makeRequestOptionsLegacy` is not used. I kept it only for now but will be entirely deleted in this PR.

Reply:
removed in https://github.com/huggingface/huggingface.js/pull/1208/commits/46ed0a128e135f4571e479e76f61d1c3864b19aa